### DISCUSSION TITLE
Unknown return value of column Number(17,2)

### DISCUSSION THREAD

**Discussion by iamjavadev1** [averemee-si/oracdc#50] [Q&A] [answered]
Hi Averemee-si, 

               I have table that contains column type is NUMBER(17,2) and when view return value in kafka topic, that value is not in number format. How can I transform it into number format?

               SAL (NUMBER(17,2)).                KAFKA
                ---------------------------------------
               1,100                              Aa2w

Regards,
Iamjavadev1

**Comment by averemee-si** [comment, marked as answer]
Hi iamjavadev1
Connector documentation clearly state in section [Oracle RDBMS Type Mapping](https://github.com/averemee-si/oracdc#oracle-rdbms-type-mapping) that by default
<!DOCTYPE html>

Oracle RDBMS Type | JSON Type | Comment
-- | -- | --
DATE | int32 | org.apache.kafka.connect.data.Date
TIMESTAMP% | int64 | org.apache.kafka.connect.data.Timestamp
NUMBER | int8 | NUMBER(1,0) & NUMBER(2,0)
NUMBER | int16 | NUMBER(3,0) & NUMBER(4,0)
NUMBER | int32 | NUMBER(5,0) & NUMBER(6,0) & NUMBER(7,0) & NUMBER(8,0)
NUMBER | int64 | Other Integers between billion and 1,000,000,000,000,000,000
NUMBER | float64 | Oracle NUMBER without specified SCALE and PRECISION
NUMBER | bytes | org.apache.kafka.connect.data.Decimal - all other numerics

So **oracdc** sends this data as [org.apache.kafka.connect.data.Decimal](https://github.com/apache/kafka/blob/trunk/connect/api/src/main/java/org/apache/kafka/connect/data/Decimal.java) and all Kafka Connect Sink connectors understand it.

If you need help with setting Data Pipeline, replication, Kafka usage, Oracle tuning etc, we will be happy to provide you with consulting services: just send us your project description and requirements to oracle@a2-solutions.eu and we will send you our proposal/quotation with all prices and timelines.

Regards,
Aleksei

**Reply by iamjavadev1** [reply]
Thank you for your help.
